zk/datastream/client/stream_client.go

```diff
@@ -86,6 +95,93 @@ func (c *StreamClient) IsVersion3() bool {
 func (c *StreamClient) GetEntryChan() chan interface{} {
 	return c.entryChan
 }
+
+func (c *StreamClient) GetL2BlockByNumber(blockNum uint64) (*types.FullL2Block, error) {
```

Comment:
If possible, please have a unit test for this function.

Reply:
It is a bit tricky to be honest, because `net.Pipe` implementation, that is probably the most straightforward approach to mock a network connection blocks on reading and writing (no buffering). `GetL2BlockByNumber` does both writing and reading to the provided connection instance. Do you have some draft idea on how to test it? Not sure whether having a `httptest.Server` would help (https://pkg.go.dev/net/http/httptest#Server), though still don't have a clear vision on how to do it.

Reply:
Not sure if it is useful, [here](https://github.com/0xPolygonHermez/cdk-erigon/blob/zkevm/zk/da/client_test.go) is an example usage of `httptest.Server`.
Maybe the simplest way is to create a real data stream server and feed some blocks/batches to it in the test? This will not only benefit this PR but for future changes to data stream modules as well.

Reply:
Fixed https://github.com/0xPolygonHermez/cdk-erigon/pull/886/commits/0698a90aa40367070732e5a0545df2b628ed689c